brainzutils/musicbrainz_db/release_group.py

```diff
@@ -163,3 +163,55 @@ def _get_release_groups_for_artist_query(db, artist_id, release_types):
         join(models.ArtistCreditName, models.ArtistCreditName.artist_credit_id == models.ReleaseGroup.artist_credit_id).\
         join(models.Artist, models.Artist.id == models.ArtistCreditName.artist_id).\
         filter(models.Artist.gid == artist_id).filter(models.ReleaseGroupPrimaryType.name.in_(release_types))
+
+
+def get_release_groups_for_label(label_id, release_types=None, limit=None, offset=None):
+    """Get all release groups linked to a label.
+
+    Args:
+        label_id (uuid): MBID of the label.
+        release_types (list): List of types of release groups to be fetched.
+        limit (int): Max number of release groups to return.
+        offset (int): Offset that can be used in conjunction with the limit.
+
+    Returns:
+        Tuple containing the list of dictionaries of release groups ordered by release year
+        and the total count of the release groups.
+    """
+    label_id = str(label_id)
+    includes_data = defaultdict(dict)
+    if release_types is None:
+        release_types = []
+    release_types = [release_type.lower() for release_type in release_types]
+    # map release types to their case sensitive name in musicbrainz.release_group_primary_type table in the database
+    release_types_mapping = {
+        'album': 'Album',
+        'single': 'Single',
+        'ep': 'EP',
+        'broadcast': 'Broadcast',
+        'other': 'Other'
+    }
+    release_types = [release_types_mapping[release_type] for release_type in release_types]
```

Comment:
what happens if one of the items in the release_types list isn't part of this mapping?

Reply:
The musicbrainz code performs a database query to get a list of release types and validates against that: https://github.com/metabrainz/musicbrainz-server/blob/da6780f4adb6ff6446b4d8779626ec4a09f3ba2c/lib/MusicBrainz/Server/WebService/Validator.pm#L88-L94

We should do the same, here and in the get_release_groups_for_artist method (this can be a shared function). For now let's not worry about caching this value. If we find in the future that it causes our lookups to take longer than we want, we can add in caching using the cache module.